src/geos_types.jl

```diff
@@ -74,20 +74,34 @@ end
 
 mutable struct Polygon <: AbstractGeometry
     ptr::GEOSGeom
-
+    # create polygon using GEOSGeom pointer - only makes sense if pointer points to a polygon or a linear ring to start with. 
     function Polygon(ptr::GEOSGeom)
-        polygon = new(ptr)
+        id = LibGEOS.geomTypeId(ptr)
+        if id == 3
```

Comment:
These are so called "magic numbers". I would name them as constants (maybe as Enum) outside the function.